tests/test_transforms.py

```diff
@@ -0,0 +1,38 @@
+from functools import partial
+
+import aesara.tensor as at
+from aesara.graph.unify import eval_if_etuple
+from kanren import run
+from kanren.graph import reduceo, walko
+from unification import var
+
+from aemcmc.transforms import location_scale_transform
+
+
+def test_normal_scale_loc_transform():
+    """"""
+
+    srng = at.random.RandomStream(0)
+    mu_a_rv = srng.normal(0, 1)
+    sigma_a_rv = srng.halfcauchy(1)
+    a_rv = srng.normal(mu_a_rv, sigma_a_rv, size=(10,))
+
+    mu_b_rv = srng.normal(0, 1)
+    sigma_b_rv = srng.halfcauchy(1)
+    b_rv = srng.normal(mu_b_rv, sigma_b_rv, size=(10))
+
+    mu = a_rv + b_rv
+    sigma_rv = srng.halfcauchy(5.0)
+    Y_rv = srng.cauchy(mu, sigma_rv)
+
+    q_lv = var()
+    (expr_graph,) = run(
+        1, q_lv, walko(partial(reduceo, location_scale_transform), Y_rv, q_lv)
```

Comment:
Just so you know, we probably aren't going to use `walk` any time soon, because it needs a lot of work in order to be efficient enough for real use.  That kind of work needs to be done somewhere between `kanren` and `etuples`, perhaps, but, in the meantime, we will end up relying on Aesara's graph-walking capabilities (e.g. `KanrenRelationSub`).

Reply:
So if I understand correctly we'll be walking the graph and use the current node as the in or out expression of the goal?

Reply:
See the example in [the documentation](https://aesara.readthedocs.io/en/latest/extending/graph_rewriting.html#minikanren).